packages/emitter-framework/src/python/components/class-declaration/class-declaration.tsx

```diff
@@ -0,0 +1,371 @@
+              {lines.map((line) => (
+                <>{line}</>
+              ))}
+            </List>,
+          ]}
+        />
+      );
+    }
+    // Doc provided as JSX (e.g., a prebuilt <py.ClassDoc />). Pass through unchanged.
+    else {
+      docElement = docSource as any;
+    }
+  }
+  return docElement;
+}
+
+/**
+ * Creates the class body for the class declaration.
+ * Determines if the class body should render any members/children.
+ * If it does, returns a ClassBody; otherwise, returns undefined, which will render a "pass";
+ * as the class body.
+ * @param $ - The Typekit.
+ * @param props - The props for the class declaration.
+ * @returns The class body for the class declaration, either a ClassBody or undefined.
+ */
+function createClassBody($: Typekit, props: ClassDeclarationProps, abstract: boolean) {
+  const validTypeMembers = isTypedClassDeclarationProps(props)
+    ? (() => {
+        if ($.model.is(props.type)) {
+          // For models, extract properties to render as dataclass fields
+          return Array.from($.model.getProperties(props.type).values());
+        } else {
+          // For interfaces, extract operations to render as abstract methods
+          const ops = (props.type as { operations: Map<string, Operation> }).operations;
+          return Array.from(createRekeyableMap(ops).values());
+        }
+      })()
+    : [];
+  const hasValidMember = validTypeMembers.length > 0;
+  const hasChildren = Array.isArray(props.children)
+    ? (props.children as any[]).length > 0
+    : props.children != null;
+
+  if (!(hasValidMember || hasChildren)) return undefined as any;
+
+  if (isTypedClassDeclarationProps(props)) {
+    return <ClassBody {...props} validTypeMembers={validTypeMembers} abstract={abstract} />;
+  }
+
+  // Fallback for non-typed props (shouldn't be called in practice due to early return in ClassDeclaration)
+  return (<>{props.children}</>) as any;
+}
+
+/**
+ * Creates the extends types for the class declaration.
+ * @param $ - The Typekit.
+ * @param type - The type to create the extends type for.
+ * @returns The extends types for the class declaration, or undefined for interfaces.
+ */
+function getExtendsType($: Typekit, type: Model | Interface): Children | undefined {
+  // For interfaces, return undefined because inheritance is flattened by TypeSpec
+  if (!$.model.is(type)) {
+    return undefined;
+  }
+
+  const extending: Children[] = [];
+
+  if (type.baseModel) {
+    if ($.array.is(type.baseModel)) {
+      extending.push(<TypeExpression type={type.baseModel} />);
+    } else if ($.record.is(type.baseModel)) {
+      // Record-based scenarios are not supported
+      // do nothing here.
+    } else {
+      // Use py.Reference to wrap the refkey for proper resolution
+      extending.push(<py.Reference refkey={efRefkey(type.baseModel)} />);
+    }
+  }
+
+  const indexType = $.model.getIndexType(type);
+  if (indexType) {
+    if ($.record.is(indexType)) {
```

Comment:
Can there be non-`Record` index types? My understanding was that the index type is the type of arbitrary unspecified properties (i.e. `Record`).